### PR TITLE
Remove relationship depth limit when loading models

### PR DESCRIPTION
**Pull request by laevandus**
### 🔗 Issue Links

Related [PBE-4853](https://stream-io.atlassian.net/browse/PBE-4853)

### 🎯 Goal

Remove relationship loading limit when converting models

### 🧪 Manual Testing Notes

Exploratory testing (create quoted messages by quoting quote etc)

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] Changelog is updated with new localization keys
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

[PBE-4853]: https://stream-io.atlassian.net/browse/PBE-4853?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Tests/StreamChatTests/Database/DTOs/ChannelDTO_Tests.swift [hunk 1]
```diff
@@ -1415,9 +1415,9 @@ final class ChannelDTO_Tests: XCTestCase {
         )
     }
 
-    // MARK: Max depth
+    // MARK: -
```
Comment (nuno-vieira) [inline Tests/StreamChatTests/Database/DTOs/ChannelDTO_Tests.swift:1418]: Why is the mark empty? 🤔

**Review by nuno-vieira — APPROVED** [review]
When we do a chain of quoting the previous message for about 6 times, some of the quoted messages disappear.  But this is the case for `develop` as well. So for me it is LGTM, but worth creating an issue to handle this.

**Review by martinmitrevski — APPROVED** [review]

File: Tests/StreamChatTests/Database/DTOs/ChannelDTO_Tests.swift (hunk 1 above)
Comment (laevandus) [reply Tests/StreamChatTests/Database/DTOs/ChannelDTO_Tests.swift:1418]: I'll change it

**Comment by laevandus** [comment]
> When we do a chain of quoting the previous message for about 6 times, some of the quoted messages disappear. But this is the case for `develop` as well. So for me it is LGTM, but worth creating an issue to handle this.

[PBE-5223](https://stream-io.atlassian.net/browse/PBE-5223)

**Comment by laevandus** [comment]
There can be recursion between replies and quoted message if I quote a message in a thread what has thread replies.

**Comment by laevandus** [comment]
> There can be recursion between replies and quoted message if I quote a message in a thread what has thread replies.

I don't have a simple change for the mentioned issue. It probably needs core data model changes but does not sound it is worth it just to get rid of relationship depth.

**Pull request closed without merging**